test/ContentfulClient.test.ts

```diff
@@ -0,0 +1,66 @@
+import "jest";
+import {ContentfulClient} from "../src/contentful";
+import fetch from "node-fetch";
+
+jest.mock("node-fetch", () => {
+    return jest.fn(() => {
+        return {
+            ok: true,
+            json: async () => {
+                return {}
+            }
+        }
+    })
+})
+
+const mockFetch = jest.fn(() => {
+    return {
+        ok: true,
+        json: async () => {
+            return {}
+        }
+    }
+})
+
+describe("Content with linked entities and assets", () => {
+    it("should make a request with correct client options", async () => {
+        const options = {
+            accessToken: "1234",
+            spaceId: "mockSpaceId",
+            host: "https://cdn-mr.contentful.com",
+            fetch: mockFetch,
+            headers: {testing: 123}
+        }
+
+        const path = '/test'
```

Comment:
You probably want to clone the object before comparing the values, or use constants.

What happens if we have a bug where we mess up the options and switch them on you?  How do you know the string path you built is actually what you expect?